<?xml version="1.0" ?>
<robot name="shelf">

  <link name="shelf_root">
    <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
  </link>

  <link name="wall_r">
  </link>
  <link name="wall_l">
  </link>
  <link name="wall_b">
  </link>

  <link name="wall_1">
  </link>
  <link name="wall_2">
  </link>
  <link name="wall_3">
  </link>

  <link name="board_0">
  </link>
  <link name="board_1">
  </link>
  <link name="board_2">
  </link>
  <link name="board_3">
  </link>

  <joint name="joint1" type="fixed">
    <parent link="shelf_root"/>
    <child link="wall_r"/>
    <origin rpy="0 0 0" xyz="0.0 -0.6 0.6"/>
  </joint>

  <joint name="joint2" type="fixed">
    <parent link="shelf_root"/>
    <child link="wall_l"/>
    <origin rpy="0 0 0" xyz="0.0 0.6 0.6"/>
  </joint>

  <joint name="joint3" type="fixed">
    <parent link="shelf_root"/>
    <child link="wall_b"/>
    <origin rpy="0 0 0" xyz="0.2 0.0 0.6"/>
  </joint>

  <joint name="joint4" type="fixed">
    <parent link="shelf_root"/>
    <child link="wall_1"/>
    <origin rpy="0 0 0" xyz="0.0 0.2 0.2"/>
  </joint>

  <joint name="joint5" type="fixed">
    <parent link="shelf_root"/>
    <child link="wall_2"/>
    <origin rpy="0 0 0" xyz="0.0 -0.2 0.6"/>
  </joint>

  <joint name="joint6" type="fixed">
    <parent link="shelf_root"/>
    <child link="wall_3"/>
    <origin rpy="0 0 0" xyz="0.0 0.2 1.0"/>
  </joint>

  <joint name="joint7" type="fixed">
    <parent link="shelf_root"/>
    <child link="board_0"/>
    <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
  </joint>

  <joint name="joint8" type="fixed">
    <parent link="shelf_root"/>
    <child link="board_1"/>
    <origin rpy="0 0 0" xyz="0.0 0.0 0.4"/>
  </joint>

  <joint name="joint9" type="fixed">
    <parent link="shelf_root"/>
    <origin rpy="0 0 0" xyz="0.0 0.0 0.8"/>
    <child link="board_2"/>
  </joint>

  <joint name="joint10" type="fixed">
    <parent link="shelf_root"/>
    <child link="board_3"/>
    <origin rpy="0 0 0" xyz="0.0 0.0 1.2"/>
  </joint>

</robot>